salt/modules/network.py

```diff
@@ -1153,7 +1153,10 @@ def mod_hostname(hostname):
         with salt.utils.fopen('/etc/sysconfig/network', 'w') as fh_:
             for net in network_c:
                 if net.startswith('HOSTNAME'):
-                    fh_.write('HOSTNAME={0}\n'.format(hostname))
+                    old_hostname = net.split('=', 1)[1].rstrip()
```

Comment:
for the record, this approach follows how the parser in rh_ip.py gets the hostname...